src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -163,65 +163,30 @@ internal static bool VerifyRestoreOutput(RestoreRequest request, CacheFile cache
             return true;
         }
 
-        private static bool IsPackageOnDisk(ISet<PackageIdentity> packagesChecked, IEnumerable<VersionFolderPathResolver> pathResolvers, LocalPackageFileCache packageFileCache, PackageIdentity identity)
-        {
-            // Each id/version only needs to be checked once
-            if (packagesChecked.Add(identity))
-            {
-                //  Check each package folder. These need to match the order used for restore.
-                foreach (var resolver in pathResolvers)
-                {
-                    // Verify the SHA for each package
-                    var hashPath = resolver.GetHashPath(identity.Id, identity.Version);
-                    var nupkgMetadataPath = resolver.GetNupkgMetadataPath(identity.Id, identity.Version);
-
-                    if (packageFileCache.Sha512Exists(hashPath) ||
-                        packageFileCache.Sha512Exists(nupkgMetadataPath))
-                    {
-                        return true;
-                    }
-                }
-                return false;
-            }
-            return true;
-        }
-
         /// <summary>
         /// Generates the dgspec to be used for the no-op optimization
-        /// This methods handles the deduping of tools and the ignoring of RestoreSettings
+        /// This methods handles the deduping of tools
         /// </summary>
         /// <param name="request">The restore request</param>
         /// <returns>The noop happy dg spec</returns>
         /// <remarks> Could be the same instance if no changes were made to the original dgspec</remarks>
         internal static DependencyGraphSpec GetNoOpDgSpec(RestoreRequest request)
         {
-            var dgSpec = request.DependencyGraphSpec;
-
-            if (request.Project.RestoreMetadata.ProjectStyle == ProjectStyle.DotnetCliTool || request.Project.RestoreMetadata.ProjectStyle == ProjectStyle.PackageReference)
+            if (request.Project.RestoreMetadata.ProjectStyle == ProjectStyle.DotnetCliTool)
             {
-                var uniqueName = request.DependencyGraphSpec.Restore.First();
-                dgSpec = request.DependencyGraphSpec.WithProjectClosure(uniqueName);
-
+                var dgSpec = request.DependencyGraphSpec.WithProjectClosure(request.DependencyGraphSpec.Restore.First());
```

Comment:
Previously the PR style projects used to get the dgspec with closure . Why is this not required anymore? Also why is required for the dotnetcli? 
Maybe comments will help for the future?

Reply:
There's a comment for dotnetclitool on the lines where they get nulled. 

It's not needed for PR style because the RestoreSettings were removed from the assets file write after the no-op normalization here was written. 

See no-op ignoring in: https://github.com/NuGet/NuGet.Client/commit/967f1586cb225c90f17c6abdf7dfb03ffcfbb123#diff-c6b7f00df939531ce334b386ef6922b4 - Jun 15, 2017 

See restore settings removal in: https://github.com/NuGet/NuGet.Client/commit/7b9f4c10883f06c14e44b7197103fb65c6559047#diff-f585ed8fd184026effea6796bc44f32f - Jan 19, 2018